api/v1alpha1/connection_types.go

```diff
@@ -5,14 +5,22 @@
 
 package v1alpha1
 
-import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
 
 // Connection allows users to configure connection-level settings
 type Connection struct {
 	// Limit defines limits related to connections
 	//
 	// +optional
 	Limit *ConnectionLimit `json:"limit,omitempty"`
+	// ConnectionBufferLimit provides configuration for the maximum buffer size for each incoming connection.
+	// Default: 32768 bytes.
+	//
+	// +optional
+	BufferLimit *resource.Quantity `json:"bufferLimit,omitempty"`
```

Comment:
any specific reason to use `quantity` and not `uint32` ?

Reply:
Thinking about UX, using quantity as input lets users specify sizes in units like megabytes, gigabytes, etc., just like how we use `*Duration` in some cases, could make things smoother